DashAI/back/api/api_v1/endpoints/explainers.py

```diff
@@ -0,0 +1,619 @@
+import logging
+import os
+import pickle
+from typing import Callable
+
+from dependency_injector.wiring import Provide, inject
+from fastapi import APIRouter, Depends, status
+from fastapi.exceptions import HTTPException
+from sqlalchemy import exc, select
+from sqlalchemy.orm import Session
+from typing_extensions import ContextManager
+
+from DashAI.back.api.api_v1.schemas.explainers_params import (
+    GlobalExplainerParams,
+    LocalExplainerParams,
+)
+from DashAI.back.containers import Container
+from DashAI.back.core.enums.status import ExplainerStatus
+from DashAI.back.dependencies.database.models import (
+    GlobalExplainer,
+    LocalExplainer,
+    Run,
+)
+from DashAI.back.dependencies.registry import ComponentRegistry
+
+logging.basicConfig(level=logging.DEBUG)
+log = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.get("/global")
+@inject
+async def get_global_explainers(
+    run_id: int,
+    session_factory: Callable[..., ContextManager[Session]] = Depends(
+        Provide[Container.db.provided.session]
+    ),
+):
+    """Returns the available global explanainers in the database associated with the
+    run_id.
+
+    Parameters
+    ----------
+    run_id: int
+        Run id to select the global explanations to retrieve.
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    List[dict]
+        A list of dicts containing global explainers.
+
+    Raises
+    ------
+    HTTPException
+        If there are no global explainers associated with the run_id in the DB.
+    """
+    with session_factory() as db:
+        try:
+            global_explainers = db.scalars(
+                select(GlobalExplainer).where(GlobalExplainer.run_id == run_id)
+            ).all()
+
+        except exc.SQLAlchemyError as e:
+            log.exception(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal database error",
+            ) from e
+
+    return global_explainers
+
+
+@router.get("/global/explanation/{explainer_id}")
```

Comment:
Here why it is required to have a subresource named `/explanation`? 
There is other subresource to have in mind when using the path `explainer/global`?
If the answer is no, then I recommend to delete `explanation` and use just `/global/{explainer_id}`

Reply:
This applies to all routes.